citesphere/src/main/java/edu/asu/diging/citesphere/core/repository/cache/PageRequestRepository.java

```diff
@@ -13,11 +13,15 @@
 
 public interface PageRequestRepository extends PagingAndSortingRepository<PageRequest, String> {
 
-    @EntityGraph(value="requestsWithFullCitations", type=EntityGraphType.LOAD)
-    List<PageRequest> findByUserAndObjectIdAndPageNumberAndZoteroObjectTypeAndSortByAndCollectionId(IUser user, String objectId, int page, ZoteroObjectType zoteroObjectType, String sortBy, String collectionId);
+    @EntityGraph(value = "requestsWithFullCitations", type = EntityGraphType.LOAD)
+    List<PageRequest> findByUserAndObjectIdAndPageNumberAndZoteroObjectTypeAndSortByAndCollectionId(IUser user,
+            String objectId, int page, ZoteroObjectType zoteroObjectType, String sortBy, String collectionId);
 
-    @Query(value="SELECT DISTINCT pr from PageRequest pr LEFT JOIN FETCH pr.citations c LEFT JOIN FETCH c.authors a WHERE pr.objectId = ?2 AND pr.user = ?1 AND pr.pageNumber = ?3 AND pr.zoteroObjectType = ?4")
+    @Query(value = "SELECT DISTINCT pr from PageRequest pr LEFT JOIN FETCH pr.citations c LEFT JOIN FETCH c.authors a WHERE pr.objectId = ?2 AND pr.user = ?1 AND pr.pageNumber = ?3 AND pr.zoteroObjectType = ?4")
 //    @EntityGraph(value="requestsWithFullCitations")
-    List<PageRequest> findPageRequestWithCitations(IUser user, String objectId, int page, ZoteroObjectType zoteroObjectType);
-} 
+    List<PageRequest> findPageRequestWithCitations(IUser user, String objectId, int page,
+            ZoteroObjectType zoteroObjectType);
 
+    @Query(value = "SELECT PageRequest.* FROM PageRequest INNER JOIN PageRequest_Citation ON PageRequest.id = PageRequest_Citation.PageRequest_id where PageRequest_Citation.citations_citationKey = ?1", nativeQuery = true)
+    List<PageRequest> findPageRequestByKey(String citationKey);
```

Comment:
have you tried using automatic query generation? should be easy to do and easier to maintain.

Reply:
i tried but it was not working.
So i used native query

Reply:
what did you try?

Reply:
@EntityGraph(value = "requestsWithFullCitations", type = EntityGraphType.LOAD)
    List<PageRequest> findByKey(String key);

I am not sure how to search based on Entity graph